syntax = "proto3";

package explosionBoy.IO;

enum WallType {
    NONE = 0;
    BASIC = 1;
}

message Coordinates {
    uint64 x = 1;
    uint64 y = 2;
}

message Level {
    uint64 num_cells_x = 1;
    uint64 num_cells_y = 2;

    float cell_size_x = 3;
    float cell_size_y = 4;
    float ceiling_height = 5;
    float wall_thickness = 6;

    repeated string walls = 10;
    repeated Coordinates spawn_locations = 11;
}
